### PR TITLE
Add multiple cert support

### PR DESCRIPTION
**Pull request by vakwetu**
There will likely be a need to create multiple certs for a dataplane service.  This has already been raised as a use case
for libvirt to allow the better segregation of traffic.

This PR adds this capability, so that it can easily be exploited in future.  This will need a corresponding change in the edpm-ansible.

Depends-On: https://github.com/openstack-k8s-operators/edpm-ansible/pull/653

Depends-On: https://github.com/openstack-k8s-operators/ci-framework/pull/1845

### CODE REVIEW

**Comment by vakwetu** [comment]
hold to end of beta freeze

**Review by jpodivin — CHANGES_REQUESTED** [review]
minor nit, otherwise god. Do you think it deserves blocker status?

File: pkg/deployment/cert.go [hunk 1]
```diff
@@ -265,6 +267,7 @@ func GetTLSNodeCert(ctx context.Context, helper *helper.Helper,
 // to make the secret name unique.
 // The convention we use here is "<nodeset.name>-<service>-certs-<index>", so for example,
 // openstack-epdm-nova-certs-0.
```
[the remark this answers is not in the document]
Comment (vakwetu) [reply pkg/deployment/cert.go]: done!

**Review by jpodivin — APPROVED** [review]

**Comment by rabi** [comment]
Are we adding functional/kuttl test coverage for this?

**Comment by vakwetu** [comment]
@rabi  The kuttl tests were modified to validate the case where only one cert is defined (key=default).  Also, I just added some kuttl tests to validate the case where two certs are defined (key=default, key=second)

**Review by slagle — COMMENTED** [review]
+2

**Comment by fultonj** [comment]
Can you ensure this is set correctly in ci-framework? 

https://github.com/openstack-k8s-operators/ci-framework/blob/e7210ad2020150fb63b1d006de930eee529ae601/roles/update_containers/templates/update_containers.j2#L2

**Comment by fao89** [comment]
recheck

**Comment by vakwetu** [comment]
recheck

**Comment by vakwetu** [comment]
recheck

**Comment by raukadah** [comment]
recheck

**Comment by vakwetu** [comment]
recheck

**Pull request merged**